src/evaluation/precompiled_policy.rs

```diff
@@ -141,4 +156,24 @@ mod tests {
     fn ignore_patch_version_test(#[case] metadata: Metadata) {
         assert!(has_minimum_kubewarden_version(&metadata).is_ok())
     }
+
+    #[rstest]
+    #[case(Metadata {
+        execution_mode: PolicyExecutionMode::KubewardenWapc,
+        protocol_version: Some(ProtocolVersion::V1),
+        ..Default::default()
+    })]
+    fn valid_protocol_version_test(#[case] metadata: Metadata) {
+        assert!(has_valid_protocol_version(&metadata).is_ok())
+    }
+
+    #[rstest]
+    #[case(Metadata {
+        execution_mode: PolicyExecutionMode::KubewardenWapc,
+        protocol_version: Some(ProtocolVersion::Unknown),
+        ..Default::default()
+    })]
+    fn invalid_protocol_version_test(#[case] metadata: Metadata) {
+        assert!(has_valid_protocol_version(&metadata).is_err())
+    }
```

Comment:
I meant to write them like that:

```rust
    #[rstest]
    #[case(Metadata {
        execution_mode: PolicyExecutionMode::KubewardenWapc,
        protocol_version: Some(ProtocolVersion::Unknown),
        ..Default::default()
    }, false)]
    fn validate_protocol_version_test(#[case] metadata: Metadata, #[case] is_valid: bool) {
        assert_eq!(is_valid, has_valid_protocol_version(&metadata).is_err())
    }
```

Also, write a test case for non-wapc policies.